### PR TITLE
Fix fragment validation

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compiledgraph/node.scala [hunk 1]
```diff
@@ -40,9 +40,11 @@ object node {
 
   case class EndingCustomNode(id: String, ref: String) extends Node
 
-  case class SubprocessStart(id: String, params: List[Parameter], next: Next) extends Node
+  case class SubprocessUsageStart(id: String, params: List[Parameter], next: Next) extends Node
```
Comment (arkadius) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/compiledgraph/node.scala:43]: Maybe we can rename it to Fragment if we touching this class?

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala [hunk 2]
```diff
@@ -55,21 +61,28 @@ object NodeDataValidator {
         case a: Filter => toValidationResponse(compiler.compileExpression(a.expression, validationContext, expectedType = Typed[Boolean], outputVar = None))
         case a: Variable => toValidationResponse(compiler.compileExpression(a.value, validationContext, expectedType = typing.Unknown, outputVar = Some(OutputVar.variable(a.varName))))
         case a: VariableBuilder => toValidationResponse(compiler.compileFields(a.fields, validationContext, outputVar = Some(OutputVar.variable(a.varName))))
-        case a: SubprocessOutputDefinition => toValidationResponse(compiler.compileFields(a.fields, validationContext, outputVar = Some(OutputVar.subprocess(a.outputName))))
+        case a: SubprocessOutputDefinition => toValidationResponse(compiler.compileFields(a.fields, validationContext, outputVar = None))
         case a: Switch => toValidationResponse(compiler.compileSwitch(Applicative[Option].product(a.exprVal, a.expression), outgoingEdges.collect {
           case OutgoingEdge(k, Some(NextSwitch(expression))) => (k, expression)
         }, validationContext))
         case a: SubprocessInput => SubprocessResolver(getFragment).resolveInput(a).fold(
```
Comment (arkadius) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala:68]: So SubprocessInput is just the FragmentUsage?

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala [hunk 3]
```diff
@@ -84,8 +82,8 @@ object OutputVar {
   def enricher(outputName: String): OutputVar =
     OutputVar(EnricherFieldName, outputName)
 
-  def subprocess(outputName: String): OutputVar =
-    OutputVar(SubprocessFieldName, outputName)
+  def fragmentOutput(outputName: String, varName: String): OutputVar =
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala:85]: why it is "ref.outputVariableNames"?  what ref it means?

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compiledgraph/node.scala (hunk 1 above)
Comment (mproch) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/compiledgraph/node.scala:43]: :+1:

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala (hunk 2 above)
Comment (mproch) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala:68]: yes, we could rename it like this. But maybe let's do it in next version, as this one needs migration (other places are artificial nodes created during resolution, not saved in DB/JSON)

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala (hunk 3 above)
Comment (mproch) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala:85]: I added comment, it's path to appropriate field in json model

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala (hunk 2 above)
Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala:68]: ok

**Review by arkadius — APPROVED** [review]

**Comment by arkadius** [comment]
Please add migration guide

**Pull request merged**